clients/client-common/src/main/java/org/eclipse/hono/client/util/MessagingClientProvider.java

```diff
@@ -36,7 +37,7 @@
  *
  * @param <T> The type of client to wrap.
  */
-public final class MessagingClient<T extends Lifecycle> implements Lifecycle, ServiceClient {
+public final class MessagingClientProvider<T extends MessagingClient & Lifecycle> implements Lifecycle, ServiceClient {
```

Comment:
The `T extends MessagingClient` is needed for the `setClient(T client)` method (not having the `final MessagingType type` parameter anymore), so that `client.getMessagingType()` can be called there.

Reply:
Oh yes, sure